### DISCUSSION TITLE
Private functions declared outside of the class

### DISCUSSION THREAD

**Discussion by PVGH19** [android/nowinandroid#538] [Q&A] [answered]
Hello!

I was looking at the code of TopicViewModel and I realized that the private functions are declared outside of the class. Why is that? Are there any benefits, memory-wise, code cleanup, etc related to that?

Thank you!

**Comment by alexvanyo** [comment, marked as answer]
The main benefit is that you can more easily identify what inputs are used in the method. If you have a method declared inside a class, those methods (for better and for worse) have access to all of the instance's parameters, properties, other methods.

A top-level method just has the parameters to the method, and that's it. If you want access to a particular parameter, or to call some instance method, you have to pass it in.

For that reason, top-level methods can lead to a data flow that's easier to understand and read. If you're looking at just [`newsUiState`](https://github.com/android/nowinandroid/blob/eb6dbaac0bc42e422e25cec3fd7ac2cd245d1a8d/feature/topic/src/main/java/com/google/samples/apps/nowinandroid/feature/topic/TopicViewModel.kt#LL131C13-L131C25), you can know that it will only depend on those three parameters, and not on `topicsRepository`.

**Comment by victorinojeffrey** [comment]
SAMSUNG GALAXY T-289
